examples/imageviewer/mapview-desktop/src/desktopMain/kotlin/example/map/MapStateExt.kt

```diff
@@ -0,0 +1,56 @@
+package example.map
+
+fun InternalMapState.geoLengthToDisplay(geoLength: Double): Int {
+    return (height * geoLength * scale).toInt()
+}
+
+fun InternalMapState.geoXToDisplay(x: Double): Int = geoLengthToDisplay(x - topLeft.x)
+fun InternalMapState.geoYToDisplay(y: Double): Int = geoLengthToDisplay(y - topLeft.y)
+@Suppress("unused")
+fun InternalMapState.geoToDisplay(geoPt: GeoPoint): DisplayPoint =
+    DisplayPoint(geoXToDisplay(geoPt.x), geoYToDisplay(geoPt.y))
+
+fun InternalMapState.displayLengthToGeo(displayLength: Int): Double =
+    displayLength / (scale * height)
+
+fun InternalMapState.displayLengthToGeo(pt: DisplayPoint): GeoPoint =
+    GeoPoint(displayLengthToGeo(pt.x), displayLengthToGeo(pt.y))
+
+fun InternalMapState.displayToGeo(displayPt: DisplayPoint): GeoPoint {
+    val x1 = displayLengthToGeo((displayPt.x))
+    val y1 = displayLengthToGeo((displayPt.y))
+    return topLeft + GeoPoint(x1, y1)
+}
+
+@Suppress("unused")
+val InternalMapState.minScale
+    get():Double = 1.0
+val InternalMapState.maxScale get():Double = (TILE_SIZE.toDouble() / height) * pow2(Config.MAX_ZOOM)
+
+fun pow2(x: Int): Int {
+    if (x < 0) {
+        return 0
```

Comment:
Wow, that's unexpected behaviour of pow2 function

Reply:
Renamed it to `fun fastPow2ForPositiveInt(x: Int): Int`